EIPS/eip-2031.md

```diff
@@ -0,0 +1,59 @@
+---
+eip: 2031
+title: State Rent B - Net transaction counter
+author: Alexey Akhunov (@AlexeyAkhunov)
+discussions-to: https://ethereum-magicians.org/t/eip-2031-net-transaction-counter-change-b-from-state-rent-v3-proposal/3283
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-05-15
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+Ethereum starts to track the number of transactions inside it state (for now, only number of transactions after this change is introduced, therefore
+it is called *Net* transaction count).
+It is done by incrementing a storage slot in the special contract, called State counter contract (see EIP 2029).
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+It is part of the State Rent roadmap. This particular change makes any Ethereum transaction increment the transaction counter, which is a special storage slot
+in the State counter contract. This counter will be used to populate the nonces of newly created
+non-contract accounts. This way of populating nonce ensures replay protection for accounts that were evicted and then brought back by sending ether to them.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Ethereum currently does not have a special place in the state for tracking number of transactions.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+A new field, with the location 0, is added to the state counter contract. It will eventually contain `txCount`, the total number of transactions processed up until that point.
```

Comment:
Can you explain what "a new field with location 0" means?

Reply:
Thanks, done